databroker/databroker.py

```diff
@@ -500,8 +501,4 @@ def get_fields(header):
     -------
     fields : set
     """
-    fields = set()
-    for desc in header['descriptors']:
-        for field in desc['data_keys'].keys():
-            fields.add(field)
-    return fields
+    return set(chain(list(d['data_keys'].keys()) for d in header['descriptors']))
```

Comment:
I'm not sure `chain` is the right approach here.

```python
>>> from itertools import chain
>>> list(chain([[1, 2, 3], [2, 3, 4]]))
[[1, 2, 3], [2, 3, 4]]
>>> set(list(chain([[1, 2, 3], [2, 3, 4]])))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'list'
```

But since bikeshedding is fun, lets just do a double list comprehension